### PR TITLE
Instance and dataset

### PR DESCRIPTION
**Pull request by pcattori**
# ↪️ Pull Request

This PR fleshes out the Instance and Dataset types to support a natural path to arrive at the new attributes package features.

Relates to #309 , #321 , #251 

Reviewers: PR is organized nicely by commit. [Start here](https://github.com/Datatamer/tamr-client/pull/338/commits/2fac78aaebc3dc4f8d24fa25d8a07592d50954de) and keep clicking "next" 😄 

## 💻 Examples

```python
import tamr_client as tc

s = tc.session.from_auth(tc.UsernamePasswordAuth('username', 'password'))
instance = tc.Instance(host='10.10.0.121')

dataset = tc.dataset.from_resource_id(s, instance, '1')
attr = tc.attribute.create(
    s,
    dataset,
    name='my attribute',
    is_nullable=False,
    type=tc.attributes.type_alias.GEOSPATIAL,
)
```

## ✔️ PR Todo

- [x] Added/updated testing for this change
- [x] Included links to related issues/PRs
- [x] Update relevant [docs](https://github.com/Datatamer/tamr-client/tree/master/docs) + docstrings
- [x] Update the [CHANGELOG](https://github.com/Datatamer/tamr-client/blob/master/CHANGELOG.md) under the current `-dev` version:
  - Add changelog entries under any that apply: **BREAKING CHANGES**, **NEW FEATURES**, **BUG FIXES**.
  - Changelog entry format: `[#<issue number>](<link to issue>) <change description>`

### CODE REVIEW

**Comment by keziah-tamr** [comment]
Looks pretty good to me. One comment. Could you add some usage of Instance to your example in addition to its creation? It's unclear to me how it should be used. Seems similar to Client

**Comment by pcattori** [comment]
@keziah-tamr : `test_dataset.py` shows usage of `tc.Instance` in the call to `tc.dataset.from_resource_id`. Does that address your comment?

File: tamr_client/datasets/dataset.py [hunk 1]
```diff
@@ -1,13 +1,91 @@
...
+    """Get dataset by URL
+
+    Fetches dataset from Tamr server
+
+    Args:
+        url: Dataset URL
+
+    Raises:
+        DatasetNotFound: If no dataset could be found at the specified URL.
+            Corresponds to a 404 HTTP error.
```
Comment (Lordluen) [inline tamr_client/datasets/dataset.py]: If the instance is down, wouldn't that return a 404 also?

File: tamr_client/__init__.py [hunk 2]
```diff
@@ -1,18 +1,53 @@
...
+beta_flag_value = "1"
+beta_flag = os.environ.get(beta_flag_name)
+
+if beta_flag != beta_flag_value:
+    msg = (
+        f"ERROR: 'tamr_client' package is in BETA, but you do not have the '{beta_flag_name}' environment variable set to '1'."
+        "\n\nHINT: Use 'tamr_unify_client' package instead for non-BETA features"
+        f"\nHINT: Set environment variable '{beta_flag_name}=1' to opt-in to BETA features."
+        "\n\nWARNING: Do not rely on BETA features in production workflows."
+        "\nTamr will not offer support for BETA features."
```
Comment (Lordluen) [inline tamr_client/__init__.py]: Last I checked, Alpha features are not supported. Beta features are supported, but don't have specific SLAs and bugs are expected, thus you should not run them in production.

File: docs/beta.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# BETA
+
+  **WARNING**: Do not rely on BETA features in production workflows.
+  Tamr will not offer support for BETA features.
```
[the remark this answers is not in the document]
Comment (pcattori) [reply docs/beta.md]: Changed wording to align with PM

File: tamr_client/datasets/dataset.py (hunk 1 above)
Comment (pcattori) [reply tamr_client/datasets/dataset.py]: If the instance is down, the `requests` will fail with a `ConnectionError`, not an HTTP status code.

**Comment by pcattori** [comment]
This PR was merged, but [Github is acting strangely](https://www.githubstatus.com/incidents/xp2qc958g4wt) and does not detect this PR as merged.

**Pull request closed without merging**